src/EnergyPlus/Plant/Component.cc

```diff
@@ -129,5 +129,13 @@ namespace DataPlant {
         this->compPtr->simulate(state, this->location, FirstHVACIteration, this->MyLoad, this->ON);
     }
 
+    void CompData::oneTimeInit(EnergyPlusData &state) const
+    {
+        if (this->compPtr->oneTimeInitFlag) {
```

Comment:
oneTimeInitFlag because we don't want the oneTimeInit_new called again if there are components that are common to multiple loops.